DelvUI/Interface/Jobs/SummonerHud.cs

```diff
@@ -83,171 +78,173 @@ public override void DrawJobHud(Vector2 origin, PlayerCharacter player)
-            {
-                return;
-            }
+            int stackCount = gauge.IsIfritReady ? 1 : 0;
+            BarUtilities.GetChunkedBars(Config.IfritBar, 1, stackCount, 0).Draw(origin);
+        }
 
-            PluginConfigColor color = stacks >= 16 ? Config.DemiStatusIndicatorBar.PhoenixColor : Config.DemiStatusIndicatorBar.BahamutColor;
-            int value = stacks < 8 ? 0 : 1;
+        private void DrawTitanBar(Vector2 origin)
+        {
+            SMNGauge gauge = Plugin.JobGauges.Get<SMNGauge>();
+            int stackCount = gauge.IsTitanReady ? 1 : 0;
+            BarUtilities.GetChunkedBars(Config.TitanBar, 1, stackCount, 0).Draw(origin);
+        }
 
-            BarUtilities.GetBar(Config.DemiStatusIndicatorBar, value, 1, fillColor: color)
-                .Draw(origin);
+        private void DrawGarudaBar(Vector2 origin)
+        {
+            SMNGauge gauge = Plugin.JobGauges.Get<SMNGauge>();
+            int stackCount = gauge.IsGarudaReady ? 1 : 0;
+            BarUtilities.GetChunkedBars(Config.GarudaBar, 1, stackCount, 0).Draw(origin);
         }
 
-        private void DrawDreadwyrmAetherBar(Vector2 origin)
+        private void HandleAttunementStacks(Vector2 origin, PlayerCharacter player)
         {
             SMNGauge gauge = Plugin.JobGauges.Get<SMNGauge>();
-            byte stacks = (byte)gauge.AetherFlags;
+            byte attunementStacks = gauge.Attunement;
+            string currentAttunement = gauge.IsIfritAttuned ? "Ifrit" : gauge.IsTitanAttuned ? "Titan" : gauge.IsGarudaAttuned ? "Garuda" : "None";
 
-            if (Config.DreadwyrmAetherBar.HideWhenInactive && stacks == 0)
+            switch (currentAttunement)
             {
-                return;
+                case "Ifrit":
+                    if (Config.StacksBar.Enabled && Config.StacksBar.ShowIfritStacks)
+                    {
+                        DrawStacksBar(origin, player, attunementStacks, 2, Config.StacksBar.IfritStackColor);
+                    }
+                    break;
+                case "Titan":
+                    if (Config.StacksBar.Enabled && Config.StacksBar.ShowTitanStacks)
+                    {
+                        DrawStacksBar(origin, player, attunementStacks, 4, Config.StacksBar.TitanStackColor);
+                    }
+                    break;
+                case "Garuda":
+                    if (Config.StacksBar.Enabled && Config.StacksBar.ShowGarudaStacks)
+                    {
+                        DrawStacksBar(origin, player, attunementStacks, 4, Config.StacksBar.GarudaStackColor);
+                    }
+                    break;
+                default:
+                    if (Config.StacksBar.Enabled && !Config.StacksBar.HideWhenInactive)
+                    {
+                        DrawStacksBar(origin, player, 0, 1, Config.StacksBar.FillColor);
+                    }
+                    break;
             }
+        }
```

Comment:
1) Any reason why you didn't check for `Config.StacksBar.Enabled` before even calling this method?
2) The `currentAttunement` variable is really weird. Making it a string and then doing a switch over it is weird. IMO just do some if-elses i.e.:
```
if (gauge.IsIfritAttuned && Config.StacksBar.ShowIfritStacks) 
{
    // ...
}
else if (gauge.IsTitanAttuned && Config.StacksBar.ShowTitanStacks)
{
    // ...
}
// etc...

Reply:
Brain lag is one hell of a drug. 

1. Meant to implement the check but skimped over it on my final check, added the check
2. I agree with this. rewrote this section has a chain of else ifs.